internal/mode/static/state/graph/grpcroute.go

```diff
@@ -234,3 +238,48 @@ func validateGRPCMethodMatch(
 	}
 	return allErrs
 }
+
+func validateGRPCFilter(
+	validator validation.HTTPFieldsValidator,
+	filter v1alpha2.GRPCRouteFilter,
+	filterPath *field.Path,
+) field.ErrorList {
+	var allErrs field.ErrorList
+
+	switch filter.Type {
+	case v1alpha2.GRPCRouteFilterRequestHeaderModifier:
+		return validateFilterHeaderModifier(validator, filter.RequestHeaderModifier, filterPath)
+	default:
+		valErr := field.NotSupported(
+			filterPath.Child("type"),
+			filter.Type,
+			[]string{
+				string(v1alpha2.GRPCRouteFilterRequestHeaderModifier),
+			},
+		)
+		allErrs = append(allErrs, valErr)
+		return allErrs
+	}
+}
+
+// convertGRPCFilters converts GRPCRouteFilters (a subset of HTTPRouteFilter) to HTTPRouteFilters
+// so we can reuse the logic from HTTPRoute filter validation and processing
+func convertGRPCFilters(filters []v1alpha2.GRPCRouteFilter) []v1.HTTPRouteFilter {
+	if len(filters) == 0 {
+		return nil
+	}
+	httpFilters := make([]v1.HTTPRouteFilter, 0, len(filters))
+	for _, filter := range filters {
+		switch filter.Type {
+		case v1alpha2.GRPCRouteFilterRequestHeaderModifier:
+			httpRequestHeaderFilter := v1.HTTPRouteFilter{
+				Type:                  v1.HTTPRouteFilterRequestHeaderModifier,
+				RequestHeaderModifier: filter.RequestHeaderModifier,
+			}
+			httpFilters = append(httpFilters, httpRequestHeaderFilter)
+		default:
+			continue
```

Comment:
Just a heads up, codecov is complaining about this line being uncovered. Can we modify the test case to cover it?